lib/Db/TimelineQueryPeopleFaceRecognition.php

```diff
@@ -214,6 +162,113 @@ public function getFaceRecognitionPreview(TimelineRoot &$root, $currentModel, $p
         return $previews;
     }
 
+    private function getFaceRecognitionClusters(TimelineRoot &$root, int $currentModel, bool $show_singles = false, bool $show_hidden = false)
+    {
+        $query = $this->connection->getQueryBuilder();
+
+        // SELECT all face clusters
+        $count = $query->func()->count($query->createFunction('DISTINCT m.fileid'));
+        $query->select('frp.id')->from('facerecog_persons', 'frp');
+        $query->selectAlias($count, 'count');
+        $query->selectAlias('frp.user', 'user_id');
+
+        // WHERE there are faces with this cluster
+        $query->innerJoin('frp', 'facerecog_faces', 'frf', $query->expr()->eq('frp.id', 'frf.person'));
+
+        // WHERE faces are from images.
+        $query->innerJoin('frf', 'facerecog_images', 'fri', $query->expr()->eq('fri.id', 'frf.image'));
+
+        // WHERE these items are memories indexed photos
+        $query->innerJoin('fri', 'memories', 'm', $query->expr()->andX(
+            $query->expr()->eq('fri.file', 'm.fileid'),
+            $query->expr()->eq('fri.model', $query->createNamedParameter($currentModel)),
+        ));
+
+        // WHERE these photos are in the user's requested folder recursively
+        $query = $this->joinFilecache($query, $root, true, false);
+
+        // GROUP by ID of face cluster
+        $query->groupBy('frp.id');
+        $query->addGroupBy('frp.user');
```

Comment:
This particular line doesn't make sense to me, but it seems that in strict mode, mariadb doesn't like that all rows ar equal and forces grouping for it as well. 🤔

Reply:
The integration with recognize in strict mode also fails with:

```
[index] Error: OC\DB\Exceptions\DbalException: An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1055 'nextcloud.rfc.user_id' isn't in GROUP BY at <<closure>>

0. /var/www/html/nextcloud/lib/private/DB/ConnectionAdapter.php line 69
   OC\DB\Exceptions\DbalException::wrap(Doctrine\DBAL\Ex ... "})
1. /var/www/html/nextcloud/apps/memories/lib/Db/TimelineQueryDays.php line 328
   OC\DB\ConnectionAdapter->executeQuery("WITH RECURSIVE  ... C", [[44623],false], [101,5])
2. /var/www/html/nextcloud/apps/memories/lib/Db/TimelineQueryPeopleRecognize.php line 75
   OCA\Memories\Db\TimelineQuery->executeQueryWithCTEs(OC\DB\QueryBuilder\QueryBuilder {})
3. /var/www/html/nextcloud/apps/memories/lib/Controller/PeopleController.php line 59
   OCA\Memories\Db\TimelineQuery->getPeopleRecognize(OCA\Memories\Db\TimelineRoot {})
4. /var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 225
   OCA\Memories\Controller\PeopleController->recognizePeople()
5. /var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php line 133
   OC\AppFramework\Http\Dispatcher->executeController(OCA\Memories\Con ... {}, "recognizePeople")
6. /var/www/html/nextcloud/lib/private/AppFramework/App.php line 172
   OC\AppFramework\Http\Dispatcher->dispatch(OCA\Memories\Con ... {}, "recognizePeople")
7. /var/www/html/nextcloud/lib/private/Route/Router.php line 298
   OC\AppFramework\App::main("OCA\\Memories\\ ... r", "recognizePeople", OC\AppFramework\ ... {}, ["memories.People.recognizePeople"])
8. /var/www/html/nextcloud/lib/base.php line 1047
   OC\Route\Router->match("/apps/memories/api/recognize/people")
9. /var/www/html/nextcloud/index.php line 36
   OC::handleRequest()

GET /apps/memories/api/recognize/people
from 190.48.188.12 by user at 2023-02-03T15:21:38+00:00
```

So, if it was never reported before, it is probably an option that I activate in mariadb, and this line is not necessary.. 🤔